### DISCUSSION TITLE
Programming environment

### DISCUSSION THREAD

**Discussion by DennisCor** [gfunkbus76/Digifiz-Dash#5] [General]
Hi Gavin
Nice Projekt your digital Dashboard
I'm working on something similar.
I don't want to replace whole dashboard, I just would like to monitore some engine parameters like oilpresure and temperature.
I also got a TPMS and GPS working.
Like you, I decidet to use Arduino controlers to read out the diferent sensors. The microcontroler send the Data over Wifi to the Raspberry MQTT Server.
For a blackbox funktion I wrote everithing in a InfluxDB Database.
To visualise everithing I fist user Grafana, this looks nice but is much to slow. So I switsht to Node Red. But it ist to limited to create a nice looking Dashboard. 
So my next step would be PyGame. 
It seams in your Video "Pre-OBD - DIY Digital Dashboard..." you are using a windows PC for programming is that right ? Wich Software are you using for programm python and animate the pygame screen ?
Like shown in the first 40sec in your Video ? 

Looking forwoard to more pics and videos of your Dash and your Bus.
Chears Dennis

**Comment by gfunkbus76** [comment]
Hi Dennis!

Sounds like a fun project. 

I work on the project in my downtime at work(on Windows), at home on either Windows or MacBook. So I bounce around a lot. I have setup node red and mqtt to run on these systems for my testing.

Once I'm more content with the software side I will roll it out onto my raspberry Pi 4. 

Software I use: 
Adobe XD for original graphics design
Arduino IDE for Arduino code
Pycharm community edition for Python/pygame editing

What you see in the video is me running the main.py file from my script, it then rinse the pygame animations etc.

Node red feeds transfers data from Arduino to MQTT. 

Hope that kinda helps! I'm going to try and do a more tutorial or project overview video when it's a little more polished. You guys are just so along for my iterative development cycle lol...
